From 8ff19c2b2f288d6851b444dbda39544980837b73 Mon Sep 17 00:00:00 2001
From: Thomas Hellstrom <thellstrom@vmware.com>
Date: Wed, 1 Feb 2012 16:59:13 +0100
Subject: [PATCH 1/4] vmwgfx: Avoid including a library header and use pixman
 for type conversion

The PictTransform type is a typedef of a pixman type.

Signed-off-by: Thomas Hellstrom <thellstrom@vmware.com>
---
 vmwgfx/vmwgfx_xa_composite.c |   20 ++++++++++----------
 1 files changed, 10 insertions(+), 10 deletions(-)

diff --git a/vmwgfx/vmwgfx_xa_composite.c b/vmwgfx/vmwgfx_xa_composite.c
index a2e3970..32246e8 100644
--- a/vmwgfx/vmwgfx_xa_composite.c
+++ b/vmwgfx/vmwgfx_xa_composite.c
@@ -33,8 +33,8 @@
 #include "config.h"
 #endif
 
+#include <pixman.h>
 #include <picturestr.h>
-#include <X11/extensions/Xrender.h>
 #include "xa_composite.h"
 #include "vmwgfx_saa.h"
 #include "vmwgfx_saa_priv.h"
@@ -75,17 +75,17 @@ vmwgfx_matrix_from_pict_transform(PictTransform *trans, float *matrix)
    if (!trans)
       return FALSE;
 
-   matrix[0] = XFixedToDouble(trans->matrix[0][0]);
-   matrix[3] = XFixedToDouble(trans->matrix[0][1]);
-   matrix[6] = XFixedToDouble(trans->matrix[0][2]);
+   matrix[0] = pixman_fixed_to_double(trans->matrix[0][0]);
+   matrix[3] = pixman_fixed_to_double(trans->matrix[0][1]);
+   matrix[6] = pixman_fixed_to_double(trans->matrix[0][2]);
 
-   matrix[1] = XFixedToDouble(trans->matrix[1][0]);
-   matrix[4] = XFixedToDouble(trans->matrix[1][1]);
-   matrix[7] = XFixedToDouble(trans->matrix[1][2]);
+   matrix[1] = pixman_fixed_to_double(trans->matrix[1][0]);
+   matrix[4] = pixman_fixed_to_double(trans->matrix[1][1]);
+   matrix[7] = pixman_fixed_to_double(trans->matrix[1][2]);
 
-   matrix[2] = XFixedToDouble(trans->matrix[2][0]);
-   matrix[5] = XFixedToDouble(trans->matrix[2][1]);
-   matrix[8] = XFixedToDouble(trans->matrix[2][2]);
+   matrix[2] = pixman_fixed_to_double(trans->matrix[2][0]);
+   matrix[5] = pixman_fixed_to_double(trans->matrix[2][1]);
+   matrix[8] = pixman_fixed_to_double(trans->matrix[2][2]);
 
    return TRUE;
 }
-- 
1.7.8.3

